src/utils.h

```diff
@@ -0,0 +1,36 @@
+/* structs.h
+
+Copyright 2021      Christopher Nelson <christopher.nelson@languidnights.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to
+deal in the Software without restriction, including without limitation the
+rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+sell copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies of the Software and its documentation and acknowledgment shall be
+given in the documentation and software packages that this Software was
+used.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+*/
+
+#ifndef SCROT_UTILS_H
+#define SCROT_UTILS_H
+
+#include <string.h>
+
+#ifndef HAVE_STRDUP
+#define strdup(a) _strdup(a)
```

Comment:
Hi @languidnights 
Some changes are needed: 
1) use the correct macro:`AC_CHECK_FUNCS` and `SCROT_HAVE_STRDUP`
2) prevent the function `_strdup` from being included in the binary if `#define SCROT_HAVE_STRDUP 1`:
```bash
nm src/scrot  | grep strdup
0000000000007900 T _strdup                                                                                                                                                                        
                 U strdup@GLIBC_2.2.5 
```
  what we want is a result like this, as I already said if `#define SCROT_HAVE_STRDUP 1`
```bash
nm src/scrot  | grep strdup
                 U strdup@GLIBC_2.2.5
```


 3) in case of failure `malloc` call to `exit(EXIT_FAILURE)` (giblib do that)

I will send a PR to your branch with the changes